diff -urN policy-1.19.1.org/file_contexts/program/apache.fc policy-1.19.1/file_contexts/program/apache.fc
--- policy-1.19.1.org/file_contexts/program/apache.fc	2004-11-09 19:35:29.000000000 +0100
+++ policy-1.19.1/file_contexts/program/apache.fc	2004-11-13 14:08:02.582707210 +0100
@@ -6,6 +6,7 @@
 /var/www/perl(/.*)?		system_u:object_r:httpd_sys_script_exec_t
 /var/www/icons(/.*)?		system_u:object_r:httpd_sys_content_t
 /var/cache/httpd(/.*)?		system_u:object_r:httpd_cache_t
+/var/cache/apache(/.*)?		system_u:object_r:httpd_cache_t
 /etc/httpd		-d	system_u:object_r:httpd_config_t
 /etc/httpd/conf.*		system_u:object_r:httpd_config_t
 /etc/httpd/logs			system_u:object_r:httpd_log_t
@@ -16,6 +17,7 @@
 /usr/lib(64)?/apache2/modules(/.*)?	system_u:object_r:httpd_modules_t
 /usr/lib(64)?/httpd(/.*)?		system_u:object_r:httpd_modules_t
 /usr/sbin/httpd		--	system_u:object_r:httpd_exec_t
+/usr/sbin/httpd\..*	--	system_u:object_r:httpd_exec_t
 /usr/sbin/apache(2)?	--	system_u:object_r:httpd_exec_t
 /usr/sbin/suexec	--	system_u:object_r:httpd_suexec_exec_t
 /usr/lib(64)?/cgi-bin/(nph-)?cgiwrap(d)? -- system_u:object_r:httpd_suexec_exec_t
